### PR TITLE
Fast egress server draining

### PR DESCRIPTION
**Pull request by samsymons**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/1203137811378537/1207555842602667/f
iOS PR: https://github.com/duckduckgo/iOS/pull/2945
macOS PR: https://github.com/duckduckgo/macos-browser/pull/2858
What kind of version bump will this require?: Major

**Description**:

This PR adds support for fast egress server draining. It works as follows:

1. Every 5 minutes, the client makes a call to the server status endpoint for the server it is connected to
2. If the response indicates that the server wants clients to transfer to another server, we will force a tunnel configuration update

**Steps to test this PR**:
1. See client PRs

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16
* [ ] macOS 10.15
* [ ] macOS 11
* [ ] macOS 12

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/NetworkProtection/PacketTunnelProvider.swift [hunk 1]
```diff
@@ -687,10 +696,12 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                     try await self.handleAdapterStarted(startReason: startReason)
 
                     // Enable Connect on Demand when manually enabling the tunnel on iOS 17.0+.
+#if os(iOS)
```
Comment (samsymons) [inline Sources/NetworkProtection/PacketTunnelProvider.swift:699]: This probably shouldn't be in this PR but it's such a tiny change that I included it here. This is to disable the iOS connect on demand workaround on macOS entirely - right now it technically executes.

File: Sources/NetworkProtection/PacketTunnelProvider.swift [hunk 2]
```diff
@@ -1421,6 +1458,7 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
         await self.tunnelFailureMonitor.stop()
         await self.latencyMonitor.stop()
         await self.entitlementMonitor.stop()
+        await self.serverStatusMonitor.stop()
```
Comment (quanganhdo) [inline Sources/NetworkProtection/PacketTunnelProvider.swift:1461]: Should we also stop this while `sleep()`?

**Review by quanganhdo — APPROVED** [review]
Code LGTM, approved with 2 small comments.

File: Sources/NetworkProtection/PacketTunnelProvider.swift (hunk 2 above)
Comment (samsymons) [reply Sources/NetworkProtection/PacketTunnelProvider.swift:1461]: Yep, thanks for the reminder, I overlooked that we don't call this function when sleeping. Perhaps we should 🤔

**Pull request merged**